templates/common/_base/files/nmstate-configuration.yaml

```diff
@@ -0,0 +1,29 @@
+mode: 0755
+path: "/usr/local/bin/nmstate-configuration.sh"
+contents:
+  inline: |
+    #!/bin/bash
+    set -eux
+
+    if [ -e /etc/nmstate/openshift/applied ]; then
+      # Already done, don't apply again
+      exit 0
+    fi
+
+    hostname=$(hostname -s)
+    hostname_file="/etc/nmstate/openshift/${hostname}.yml"
```

Comment:
just curious, is it possible to add everything into one file and post process it later?

i.e. instead of having ip1.yaml ip2.yaml cluster.yaml, I have one big config.yaml, that then I specify within:
ip1.spec and ip2.spec

or something like that so it's a singular path, and the script itself processes when it does the copy/processing?

Reply:
There's probably some way we could do it. I'm not sure we have yq available on the host so I don't know if we can use that, but we could probably split the file with some bash magic.